flink-clients/src/test/java/org/apache/flink/client/CliFrontendStopTest.java

```diff
@@ -0,0 +1,157 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.client;
+
+import java.util.UUID;
+
+import akka.actor.*;
+import akka.testkit.JavaTestKit;
+
+import org.apache.flink.client.cli.CommandLineOptions;
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.instance.ActorGateway;
+import org.apache.flink.runtime.instance.AkkaActorGateway;
+import org.apache.flink.runtime.messages.JobManagerMessages;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import scala.Option;
+import static org.apache.flink.client.CliFrontendTestUtils.pipeSystemOutToNull;
+import static org.junit.Assert.*;
+
+public class CliFrontendStopTest {
+
+	private static ActorSystem actorSystem;
+
+	@BeforeClass
+	public static void setup() {
+		pipeSystemOutToNull();
+		actorSystem = ActorSystem.create("TestingActorSystem");
+	}
+
+	@AfterClass
+	public static void teardown() {
+		JavaTestKit.shutdownActorSystem(actorSystem);
+		actorSystem = null;
+	}
+
+	@BeforeClass
+	public static void init() {
+		CliFrontendTestUtils.pipeSystemOutToNull();
+		CliFrontendTestUtils.clearGlobalConfiguration();
+	}
+
+	@Test
+	public void testStop() {
+		try {
+			// test unrecognized option
+			{
+				String[] parameters = { "-v", "-l" };
+				CliFrontend testFrontend = new CliFrontend(CliFrontendTestUtils.getConfigDir());
+				int retCode = testFrontend.stop(parameters);
+				assertTrue(retCode != 0);
+			}
+
+			// test missing job id
+			{
+				String[] parameters = {};
+				CliFrontend testFrontend = new CliFrontend(CliFrontendTestUtils.getConfigDir());
+				int retCode = testFrontend.stop(parameters);
+				assertTrue(retCode != 0);
+			}
+
+			// test stop properly
+			{
+				JobID jid = new JobID();
+				String jidString = jid.toString();
+
+				final Option<UUID> leaderSessionID = Option.<UUID>apply(UUID.randomUUID());
+				final ActorRef jm = actorSystem.actorOf(Props.create(CliJobManager.class, jid));
+
+				final ActorGateway gateway = new AkkaActorGateway(jm, leaderSessionID);
+
+				String[] parameters = { jidString };
+				StopTestCliFrontend testFrontend = new StopTestCliFrontend(gateway);
+
+				int retCode = testFrontend.stop(parameters);
+				assertTrue(retCode == 0);
+			}
+
+			// test stop properly
+			{
+				JobID jid1 = new JobID();
+				JobID jid2 = new JobID();
+
+				final Option<UUID> leaderSessionID = Option.<UUID>apply(UUID.randomUUID());
+				final ActorRef jm = actorSystem.actorOf(Props.create(CliJobManager.class, jid1));
+
+				final ActorGateway gateway = new AkkaActorGateway(jm, leaderSessionID);
+
+				String[] parameters = { jid2.toString() };
+				StopTestCliFrontend testFrontend = new StopTestCliFrontend(gateway);
+
+				assertTrue(testFrontend.stop(parameters) != 0);
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+			fail("Program caused an exception: " + e.getMessage());
```

Comment:
Why do you print the stack trace and let the test fail with an `AssertionError` instead of the original exception?